### ISSUE TITLE
Util pod flooded with rsync's  `connect from UNDETERMINED` logs

### ISSUE THREAD

**Issue by ghost** [garden-io/garden#4242]
### Current Behavior

Util pod logs every second with `connect from UNDETERMINED` from rsync.

```
k logs pod/garden-util-59f5bd8f9-cfmz

>2023/05/02 16:26:42 [1] rsyncd version 3.2.5 starting, listening on port 8730
>2023/05/02 16:26:43 [10] connect from UNDETERMINED (10.0.28.160)
>2023/05/02 16:26:44 [11] connect from UNDETERMINED (10.0.28.160)
>2023/05/02 16:26:44 [12] connect from UNDETERMINED (10.0.28.160)
>2023/05/02 16:26:45 [13] connect from UNDETERMINED (10.0.28.160)
>2023/05/02 16:26:46 [14] connect from UNDETERMINED (10.0.28.160)
>2023/05/02 16:26:47 [15] connect from UNDETERMINED (10.0.28.160)
>2023/05/02 16:26:48 [16] connect from UNDETERMINED (10.0.28.160)
```

### Expected behavior

Rsync's reverse lookup is disabled as expected and these logs don't show up. Ensuring rsync's [config](https://github.com/garden-io/garden/blob/main/images/k8s-util/rsync-server.sh#L14) works as desired.

### Reproducible example

Any example with remote util container will stream similar logs

### Workaround

N/A

### Your environment

* **OS:** macOS 13.2.1
* **How I'm running Kubernetes:** docker-desktop 

`garden version`
0.12.55

**Comment by edvald** [comment]
This is almost definitely the liveness probe :)

**Comment by stefreak** [comment]
Is this a bug? Rsync is just logging the incoming connections, probably from the liveness probe as @edvald pointed out. What exactly is the behaviour you expect?

**Comment by stefreak** [comment]
If the problem is the amount of messages, we could try to tweak the `periodSeconds` of the liveness probe.

**Comment by worldofgeese** [comment]
@fabianboerner's code sync stopped working (files cease syncing) and his logs are flooded with the same error. Is there an update on this issue?

```console
mkdir -p /.garden/hot_reload/root/app/src/campaign-management && mkdir -p /.garden/hot_reload/tmp/app/src/campaign-management/campaign-publishing-control/ && cp -r /app/src/campaign-management/campaign-publishing-control/ /.garden/hot_reload/root/app/src/campaign-management/campaign-publishing-control/ && mkdir -p /.garden/hot_reload/root/app/src/packages && mkdir -p /.garden/hot_reload/tmp/app/src/packages/common/ && cp -r /app/src/packages/common/ /.garden/hot_reload/root/app/src/packages/common/"


seems this was executed but /.garden/ does not exit in the container
2023/06/05 23:21:29 [549] connect from UNDETERMINED (10.244.1.178)
2023/06/05 23:21:30 [550] connect from UNDETERMINED (10.244.1.178)
2023/06/05 23:21:31 [551] connect from UNDETERMINED (10.244.1.178)
2023/06/05 23:21:32 [552] connect from UNDETERMINED (10.244.1.178)
2023/06/05 23:21:33 [553] connect from UNDETERMINED (10.244.1.178)
2023/06/05 23:21:34 [554] connect from UNDETERMINED (10.244.1.178)
2023/06/05 23:21:35 [555] connect from UNDETERMINED (10.244.1.178)
2023/06/05 23:21:36 [556] connect from UNDETERMINED (10.244.1.178)
2023/06/05 23:21:37 [557] connect from UNDETERMINED (10.244.1.178)

garden-rsync container
Execing '/home/fabianboerner/.garden/tools/mutagen/c8f927f6b7e9b6c5/mutagen sync list --template={{ json . }}' in /tmp/tmp-23165-QkCTjLfkkmLL
```

**Comment by worldofgeese** [comment]
Placeholder to do Discord link

**Comment by worldofgeese** [comment]
Discord conversation [is here](https://discord.com/channels/817392104711651328/1115413317343781035)

**Comment by 10ko** [comment]
This should be fixed in `0.13.x`, please re-open if that's still happening.
